frontend/lib/src/components/elements/Expander/styled-components.ts

```diff
@@ -49,49 +56,71 @@ export const StyledDetails = styled.details<StyledDetailsProps>(
   })
 )
 
-export const StyledSummaryHeading = styled.span(({ theme }) => ({
+export const StyledSummaryHeading = styled.span({
   display: "flex",
-  gap: theme.spacing.sm,
   alignItems: "center",
   flexGrow: 1,
-}))
+})
 
 interface StyledSummaryProps {
   isStale: boolean
+  expanded: boolean
 }
 
 export const StyledSummary = styled.summary<StyledSummaryProps>(
-  ({ theme, isStale }) => ({
+  ({ theme, isStale, expanded }) => ({
     position: "relative",
     display: "flex",
     width: "100%",
+    "&:focus": {
+      outline: "none",
+    },
     "&:focus-visible": {
-      outline: `${theme.sizes.borderWidth} solid ${theme.colors.primary}`,
-      outlineOffset: `-${theme.sizes.borderWidth}`,
-      borderRadius: theme.radii.default,
+      boxShadow: `0 0 0 0.2rem ${transparentize(theme.colors.primary, 0.5)}`,
     },
-    fontSize: theme.fontSizes.sm,
-    paddingLeft: theme.spacing.lg,
-    paddingRight: theme.spacing.lg,
-    paddingTop: theme.spacing.md,
-    paddingBottom: theme.spacing.md,
+    fontSize: "inherit",
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingTop: theme.spacing.twoXS,
+    paddingBottom: theme.spacing.twoXS,
+    minHeight: theme.sizes.minElementHeight,
+    alignItems: "center",
     cursor: "pointer",
     listStyleType: "none",
     "&::-webkit-details-marker": {
       display: "none",
     },
-    "&:hover": {
-      color: theme.colors.primary,
+    backgroundColor: expanded ? theme.colors.bgMix : "transparent",
+    // When expanded, only round the top corners
+    borderRadius: expanded
+      ? `${theme.radii.default} ${theme.radii.default} 0 0`
+      : theme.radii.default,
+    // Animate border-radius changes when expanding/collapsing to match the animation of
+    // the expander content. Use a delay when collapsing because the content first needs
+    // to slide up.
+    transition: expanded
+      ? `border-radius 200ms cubic-bezier(0.23, 1, 0.32, 1), background-color 150ms ease`
+      : `border-radius 200ms cubic-bezier(0.23, 1, 0.32, 1) 300ms, background-color 150ms ease`,
+    "&:hover, &:focus-visible": {
+      backgroundColor: theme.colors.darkenedBgMix15,
     },
-    "&:hover svg": {
-      fill: theme.colors.primary,
+    "&:active": {
+      backgroundColor: theme.colors.darkenedBgMix25,
     },
     ...(isStale && STALE_STYLES),
   })
 )
 
 export const StyledDetailsPanel = styled.div(({ theme }) => ({
-  paddingBottom: theme.spacing.lg,
-  paddingLeft: theme.spacing.lg,
-  paddingRight: theme.spacing.lg,
+  padding: theme.spacing.lg,
+  borderTop: `${theme.sizes.borderWidth} solid ${theme.colors.borderColor}`,
+}))
+
+export const StyledStatusSpinner = styled(
+  StyledSpinnerIcon as React.ComponentType<StyledSpinnerIconProps>
+)(({ theme }: { theme: EmotionTheme }) => ({
+  // Make the spinner in `st.status` a bit thinner and grayscale to look
+  // similar to a material icon.
+  borderTopColor: theme.colors.bodyText,
```

Comment:
Note that I might also apply this styling to `st.spinner` in general later on, but wanted to tackle that in a separate PR.